python_modules/libraries/dagster-embedded-elt/dagster_embedded_elt/sling/asset_decorator.py

```diff
@@ -0,0 +1,181 @@
+import re
+from typing import Any, Callable, Iterable, Mapping, Optional
+
+from dagster import (
+    AssetKey,
+    AssetsDefinition,
+    AssetSpec,
+    BackfillPolicy,
+    FreshnessPolicy,
+    PartitionsDefinition,
+    multi_asset,
+)
+from dagster._annotations import public
+from dagster._core.definitions.asset_dep import CoercibleToAssetDep
+from dagster._core.definitions.auto_materialize_policy import AutoMaterializePolicy
+
+from dagster_embedded_elt.sling.sling_replication import SlingReplicationParam, validate_replication
+
+
+def get_streams_from_replication(replication: Mapping[str, Any]) -> Iterable[str]:
+    """Returns the set of streams from a Sling replication config."""
+    keys = []
+
+    for key, value in replication["streams"].items():
+        if isinstance(value, dict):
+            asset_key_config = value.get("meta", {}).get("dagster", {}).get("asset_key", [])
+            keys.append(asset_key_config if asset_key_config else key)
+        else:
+            keys.append(key)
+    return keys
+
+
+class DagsterSlingTranslator:
+    @classmethod
+    def sanitize_stream_name(cls, stream_name: str) -> str:
```

Comment:
Does this need to be a classmethod? Are we also expecting users to override this?

Reply:
I wanted to give them the option, for example, maybe they want to skip underscores but otherwise keep the implementation on the translators the same.